jfix-armeria-facade/src/main/kotlin/ru/fix/armeria/facade/webclient/impl/BaseHttpClientBuilderImpl.kt

```diff
@@ -242,25 +250,64 @@ internal abstract class BaseHttpClientBuilderImpl<out HttpClientBuilderT : BaseH
         baseBuilderState.copy(sessionProtocol = sessionProtocol)
     )
 
-    override fun setClientFactory(clientFactory: ClientFactory): HttpClientBuilderT = copyOfThisBuilder(
-        baseBuilderState.copy(clientFactory = clientFactory)
-    )
+    override fun setClientFactory(clientFactory: ClientFactory): HttpClientBuilderT {
+        if (baseBuilderState.clientFactoryBuilderCustomizers.isNotEmpty()) {
+            logger.warn {
+                """ClientFactory is set when there are some clientFactoryBuilderCustomizers specified:
+                    |- current builder state: $baseBuilderState
+                    |- provided clientFactory: $clientFactory
+                """.trimMargin()
+            }
+        }
+        return copyOfThisBuilder(
+            baseBuilderState.copy(clientFactory = clientFactory)
+        )
+    }
 
     override fun customizeArmeriaClientFactoryBuilder(
-        customizer: ClientFactoryBuilder.() -> ClientFactoryBuilder
+        customizer: ClientFactoryBuilderCustomizer
+    ): HttpClientBuilderT {
+        logWarnIfClientFactoryIsSpecified(customizer)
+        return copyOfThisBuilder(
+            baseBuilderState.copy(clientFactoryBuilderCustomizers = listOf(customizer))
```

Comment:
Maybe we also should add a warn message if new customizer will override clientFactoryBuilderCustomizers (in case if list is not empty) ?

Reply:
Fixed